src/main/scala/org/broadinstitute/dsde/workbench/sam/directory/PostgresDirectoryDAO.scala

```diff
@@ -475,26 +475,91 @@ class PostgresDirectoryDAO(protected val dbRef: DbReference,
 
   override def isEnabled(subject: WorkbenchSubject): IO[Boolean] = ???
 
-  override def getUserFromPetServiceAccount(petSA: ServiceAccountSubjectId): IO[Option[WorkbenchUser]] = ???
+  override def getUserFromPetServiceAccount(petSA: ServiceAccountSubjectId): IO[Option[WorkbenchUser]] = {
+    import SamTypeBinders._
+
+    runInTransaction { implicit session =>
+      val petServiceAccountTable = PetServiceAccountTable.syntax
+
+      val userTable = UserTable.syntax
+      val userColumn = UserTable.column
+
+      val loadUserQuery = samsql"""select ${userTable.id}, ${userTable.email}, ${userTable.googleSubjectId} from ${UserTable.table} left join ${PetServiceAccountTable.table} on ${petServiceAccountTable.userId} = ${userTable.id} where ${petServiceAccountTable.googleSubjectId} = ${GoogleSubjectId(petSA.value)}""" //TODO harmonize the subjectId types
```

Comment:
note on this todo: We could replace `GoogleSubjectId(petSA.value)` with just `petSA` and everything seems to still work (tests pass at least). Not sure if that would really address the underlying disharmony between `GoogleSubjectId` and `ServiceAccountSubjectId` that this comment is getting at, but it does make this particular query look a little nicer